File: cmd/init/greet.js
// Get arguments from environment variables
const name = process.env.ARGS_NAME || 'World';
const age = process.env.ARGS_AGE;

// Get flags from environment variables
const uppercase = process.env.FLAGS_UPPERCASE === 'true';
const lowercase = process.env.FLAGS_LOWERCASE === 'true';
const repeat = parseInt(process.env.FLAGS_REPEAT || '1');

// Construct base greeting
let greeting = `Hello ${name}!`;
if (age) {
    greeting += ` You are ${age} years old.`;
}

// Apply case transformations
if (uppercase) {
    greeting = greeting.toUpperCase();
} else if (lowercase) {
    greeting = greeting.toLowerCase();
}

// Output greeting with repetition
for (let i = 0; i < repeat; i++) {
    console.log(greeting);
}

File: cmd/init/greet.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	// Get arguments from environment variables
	name := getEnv("ARGS_NAME")
	age := getEnv("ARGS_AGE")

	// Get flags from environment variables
	uppercase := getEnv("FLAGS_UPPERCASE") == "true"
	lowercase := getEnv("FLAGS_LOWERCASE") == "true"
	repeat, _ := strconv.Atoi(getEnv("FLAGS_REPEAT"))

	// Construct base greeting
	greeting := fmt.Sprintf("Hello %s!", name)
	if age != "" {
		greeting += fmt.Sprintf(" You are %s years old.", age)
	}

	// Apply case transformations
	if uppercase {
		greeting = strings.ToUpper(greeting)
	} else if lowercase {
		greeting = strings.ToLower(greeting)
	}

	// Output greeting with repetition
	for i := 0; i < repeat; i++ {
		fmt.Println(greeting)
	}
}

func getEnv(key string) string {
	if value, exists := os.LookupEnv(key); exists {
		return value
	}
	return ""
}
